### ISSUE TITLE
Wrong type inferred when passing list concatenation expression to Iterable[T] argument

### ISSUE THREAD

**Issue by pinterior** [microsoft/pyright#6273]
**Describe the bug**

applying `f(x: Iterable[T])` to variable holding result of `[1] + ["foo"]` properly infers `T` is `int | str`.
but applying to expression `[1] + ["foo"]` infers `T` as `int`.

**Code or Screenshots**

```python
from typing import Generic, Iterable, TypeVar

T = TypeVar("T")

class A(Generic[T]):
    def __init__(self, i: Iterable[T]):
        ...

def f(i: Iterable[T]) -> T:
    ...

a = [1] + ["foo"]
reveal_type(f(a))
reveal_type(A(a))

reveal_type(f([1] + ["foo"]))
reveal_type(A([1] + ["foo"]))
```
```
tp.py
  tp.py:13:13 - information: Type of "f(a)" is "str | int"
  tp.py:14:13 - information: Type of "A(a)" is "A[str | int]"
  tp.py:16:13 - information: Type of "f([1] + ["foo"])" is "int"
  tp.py:17:13 - information: Type of "A([1] + ["foo"])" is "A[T@A | int]"
0 errors, 0 warnings, 4 informations 
```

**VS Code extension or command-line**

command-line tool, 1.1.333

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.337, which I just published. It will also be included in a future release of pylance.
